FirebaseAdmin/FirebaseAdmin/Auth/Providers/ProviderConfigClient.cs

```diff
@@ -0,0 +1,246 @@
+        /// pagination support.
+        /// </summary>
+        protected abstract class AbstractListRequest
+        : IClientServiceRequest<AuthProviderConfigs<T>>
+        {
+            private const int MaxListResults = 100;
+
+            private readonly ApiClient client;
+
+            protected AbstractListRequest(
+                ApiClient client, ListProviderConfigsOptions options)
+            {
+                this.client = client;
+                this.RequestParameters = new Dictionary<string, IParameter>();
+                this.SetPageSize(options?.PageSize);
+                this.SetPageToken(options?.PageToken);
+            }
+
+            public abstract string MethodName { get; }
+
+            public abstract string RestPath { get; }
+
+            public string HttpMethod => "GET";
+
+            public IDictionary<string, IParameter> RequestParameters { get; }
+
+            public IClientService Service { get; }
+
+            protected ApiClient ApiClient => this.client;
+
+            public HttpRequestMessage CreateRequest(bool? overrideGZipEnabled = null)
+            {
+                var query = this.RequestParameters.ToDictionary(
+                    entry => entry.Key, entry => entry.Value.DefaultValue as object);
+                return new HttpRequestMessage()
+                {
+                    Method = System.Net.Http.HttpMethod.Get,
+                    RequestUri = BuildUri(this.RestPath, query),
+                };
+            }
+
+            public async Task<Stream> ExecuteAsStreamAsync(CancellationToken cancellationToken)
+            {
+                var request = this.CreateRequest();
+                var response = await this.ApiClient.SendAsync(request, cancellationToken)
+                    .ConfigureAwait(false);
+                return await response.Content.ReadAsStreamAsync().ConfigureAwait(false);
+            }
+
+            public Stream ExecuteAsStream()
+            {
+                return this.ExecuteAsStreamAsync().Result;
+            }
+
+            public async Task<Stream> ExecuteAsStreamAsync()
+            {
+                return await this.ExecuteAsStreamAsync(default).ConfigureAwait(false);
+            }
+
+            public AuthProviderConfigs<T> Execute()
+            {
+                return this.ExecuteAsync().Result;
+            }
+
+            public async Task<AuthProviderConfigs<T>> ExecuteAsync()
+            {
+                return await this.ExecuteAsync(default).ConfigureAwait(false);
+            }
+
+            public abstract Task<AuthProviderConfigs<T>> ExecuteAsync(
+                CancellationToken cancellationToken);
+
+            internal void SetPageSize(int? pageSize)
+            {
+                if (pageSize > MaxListResults)
+                {
+                    throw new ArgumentException("Page size must not exceed 100.");
```

Comment:
nit: use string interpolation for `MaxListResults`

Reply:
Done